### DISCUSSION TITLE
HTML response in utf 8.

### DISCUSSION THREAD

**Discussion by xxl1212** [fastapi/fastapi#6697] [Questions] [answered]
Hi,

I created an index.html file that displays "Moderation" with the <p> tag.

In my main.py there is: 

from fastapi import FastAPI
from fastapi.responses import HTMLResponse
import os
app = FastAPI ()
@ app.get ("/ testHTML /", response_class = HTMLResponse)
async def read_items ():
         f = open ("index.html", "r")
         return HTMLResponse (f.read ())

But when I test I see this on the webpage:  
![image](https://user-images.githubusercontent.com/71176675/123992083-95381080-d9cb-11eb-8b32-034f7d29eaff.png)
I tried to add utf-8 meta tag to the HTML but it didn't change anything, I think it's an encoding problem in the HTMLResponse.

**Comment by dmitrijkir** [comment]
Did you try use decode?
`f.read().decode("utf-8")`

**Comment by xxl1212** [comment]
Hello,

decode generates an error (AttributeError: 'str' object has no attribute 'decode') and encode does not change anything.

Thanks anyways !

**Comment by dmitrijkir** [comment, marked as answer]
You also can try this `f = open("index.html", "r", encoding="utf-8")`

**Comment by xxl1212** [comment]
Thank you ! It works perfectly!
